examples/quantization_aware_training/torch/resnet18/main.py

```diff
@@ -0,0 +1,325 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import re
+import subprocess
+import warnings
+from pathlib import Path
+from typing import List, Optional, Tuple
+
+import openvino as ov
+import torch
+import torch.nn as nn
+import torch.nn.parallel
+import torch.optim
+import torch.utils.data
+import torch.utils.data.distributed
+import torchvision.datasets as datasets
+import torchvision.models as models
+import torchvision.transforms as transforms
+from fastdownload import FastDownload
+from torch.jit import TracerWarning
+
+import nncf
+
+BASE_MODEL_NAME = "resnet18"
+IMAGE_SIZE = 64
+BATCH_SIZE = 128
+
+
+ROOT = Path(__file__).parent.resolve()
+CHECKPOINT_URL = (
+    "https://storage.openvinotoolkit.org/repositories/nncf/openvino_notebook_ckpts/302_resnet18_fp32_v1.pth"
+)
+DATASET_URL = "http://cs231n.stanford.edu/tiny-imagenet-200.zip"
+DATASET_PATH = "~/.cache/nncf/datasets"
+
+
+def download_dataset() -> Path:
+    downloader = FastDownload(base=DATASET_PATH, archive="downloaded", data="extracted")
+    return downloader.get(DATASET_URL)
+
+
+def load_checkpoint(model: torch.nn.Module) -> torch.nn.Module:
+    checkpoint = torch.hub.load_state_dict_from_url(CHECKPOINT_URL, map_location=torch.device("cpu"), progress=False)
+    model.load_state_dict(checkpoint["state_dict"])
+    return model, checkpoint["acc1"]
+
+
+def main():
+    torch.manual_seed(0)
+    device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+    print(f"Using {device} device")
+
+    num_classes = 200  # 200 is for Tiny ImageNet, default is 1000 for ImageNet
+    init_lr = 1e-4
+
+    model = models.resnet18(pretrained=False)
```

Comment:
```suggestion
    model = models.resnet18(weights=None)
```

`UserWarning: The parameter 'pretrained' is deprecated since 0.13 and may be removed in the future, please use 'weights' instead.`